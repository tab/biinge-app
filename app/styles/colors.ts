const colors = {
  americanSilver: "#d1cecf",
  black: "#181818",
  darkBlue: "#2a3642",
  darkCandyAppleRed: "#9f0909",
  graniteGray: "#666666",
  gray: "#b5b5b5",
  grayDark: "#6e6969",
  gunmetal: "#2b2835",
  lotion: "#fafafa",
  orangeYellow: "#eec01e",
  palmLeaf: "#699d3a",
  raisinBlack: "#221f1f",
  white: "#ffffff",
}

export default colors
