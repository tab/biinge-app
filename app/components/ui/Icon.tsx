import React from "react"
import Ionicons from "react-native-vector-icons/Ionicons"

import colors from "styles/colors"

type Props = {
  name: string
  size: number
  color?: string
} & React.ComponentProps<typeof Ionicons>

const IconComponent = ({
  name,
  size,
  color = colors.white,
  ...restProps
}: Props) => {
  return <Ionicons name={name} color={color} size={size} {...restProps} />
}

export default IconComponent
