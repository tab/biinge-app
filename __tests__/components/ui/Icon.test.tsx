import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import Icon from "components/ui/Icon"
import colors from "styles/colors"

describe("Icon", () => {
  it("renders correctly", () => {
    const tree = render(
      <Icon name="star" color={colors.orangeYellow} size={20} />,
    )
    expect(tree).toMatchSnapshot()
  })
})
