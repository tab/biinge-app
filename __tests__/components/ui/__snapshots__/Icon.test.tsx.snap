// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "#eec01e",
        "fontSize": 20,
      },
      undefined,
      {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
>
  îº–
</Text>
`;
